File: examples/simple.cc
#include <iostream>
#include <openql>

int main() {
    // create platform
    auto platf = ql::Platform("seven_qubits_chip", "cc_light");

    // create program
    auto prog = ql::Program("aProgram", platf, 2);

    // create kernel
    auto k = ql::Kernel("aKernel", platf, 2);

    k.gate("prepz", 0);
    k.gate("prepz", 1);
    k.gate("x", 0);
    k.gate("y", 1);
    k.measure(0);
    k.measure(1);

    // add kernel to program
    prog.add_kernel(k);

    // compile the program
    prog.compile();

    return 0;
}

File: examples/simple.py
import openql as ql

ql.initialize()

ql.set_option('output_dir', 'output')
ql.set_option('log_level', 'LOG_INFO')

platform = ql.Platform('my_platform', 'none')

nqubits = 3
program = ql.Program('my_program', platform, nqubits)
kernel = ql.Kernel('my_kernel', platform, nqubits)

for i in range(nqubits):
    kernel.prepz(i)

kernel.x(0)
kernel.hadamard(1)
kernel.cz(2, 0)
kernel.measure(0)
kernel.measure(1)

program.add_kernel(kernel)

program.compile()
